build.gradle

```diff
@@ -132,11 +137,7 @@ task zipReport(dependsOn: 'buildFrontend', type: Zip) {
     destinationDir = file('src/main/resources')
 }
 
-tasks.shadowJar.dependsOn('zipReport');
-tasks.compileJava.dependsOn('zipReport');
-tasks.run.dependsOn('compileJava');
-
-task systemtest(dependsOn: 'zipReport', type: Test) {
+task systemtest(type: Test) {
```

Comment:
This might be a separate discussion but our systemtests currently are more of an integrated pure backend test. In addition to bypassing CLI arguments (as is an issue raised in #1645), we also don't seem to have any tests for the integration of front and backend - specifically the correct replication of `template.zip` files.

Should we also be including doing frontend-backend integration tests in systemtests as well?

Reply:
> specifically the correct replication of `template.zip` files.

I think this can be settled by changing `ConfigSystemTest` to compare the `index.html` file since right now it only compares the `.json` files. Although this would require someone to 'validate' the html file each time something is changed to changes the expected html

Reply:
For any changes to `ConfigSystemTest`, I think they are best done in a separate PR. However, if `index.html` is going to be compared, the dependency on `zipReport` should be added back in that PR

Reply:
> I think they are best done in a separate PR

Yes, I agree. I think it's also okay to leave it as-is and add back the dependency if we decide to check frontend build files.

Reply:
I just want to say that I prefer for `zipReport` to be removed completely since it is basically a convoluted way of copying files around, instead of just copying it directly to the output folder.

Reply:
From what I know, `zipReport` would still be needed to ensure that there is a `templateZip.zip` file in `src/main/resources` so that all the frontend assets in the zip file can be packaged in the jar file when running `shadowJar`.

Once a release jar file is downloaded by someone, the backend code is needed to copy the frontend assets inside to the output directory. This enables viewing the report in browser. Gradle is only relevant for users who fork the entire project.